rsocket/statemachine/RSocketStateMachine.cpp

```diff
@@ -87,8 +87,11 @@ bool RSocketStateMachine::resumeServer(
     yarpl::Reference<FrameTransport> frameTransport,
     const ResumeParameters& resumeParams) {
   if (!isDisconnectedOrClosed()) {
-    disconnect(folly::exception_wrapper(
-        std::runtime_error("A new transport is resuming the session")));
+    VLOG(1) << "Old connection (" << frameTransport_.get() << ") exists. "
```

Comment:
The bug fix here: 

Say server runs two threads T1 and T2.
client's statemachine runs on T1.
client has a resumed transport on T2.
client resumes very quickly again and connects on T1.  (the old connection on T2 hasn't been teared off yet).

A: In resumeServer() we make a check whether the old connection exists, if so we disconnect() that and connect to the new transport.   Since the old transport is running on T2, the FrameTransport::close() gets scheduled on T2 and the stateMachine is connected to the new transport.

B: Before FrameTransport::close() gets executed, the old connection gets actually teared from network and FrameTransport::cancel() gets called which in turn invokes FrameProcessor::onTerminal.  The onTerminal() gets scheduled on T1 and it ends up closing the new connection.

This happens very rarely and I was able to reproduce this only with stress testing (also the reconnections have to be really quick).

Reply:
There is a TODO comment at the end of the function, stating that the return value should not be ignored. Can we do anything for that? Should we just accept a lambda and execute it at the inside of this function for the cases that we are willing to return false?

Reply:
The return value is not used anywhere.  If resumeServer() fails it will inform the client by sending a ERROR frame or by closing the transport, else it will send a RESUME_OK.  We should change the return value (probably in another PR)